src/index.js

```diff
@@ -89,6 +89,41 @@ kalamata.expose = function(model, _opts_) {
             var beforeResult = runHooks(beforeHooks.getCollection, [req, res, mod]);
             if(res.headersSent) return;
 
+            // If we send page or page_size, we want to limit the query
+            if(req.query.page || req.query.page_size) {
+                // By default, we return the first page with 100 items
+                const { page = 1, page_size = 100 } = req.query;
+                const page_number = parseInt(page, 10);
+                const page_size_number = parseInt(page_size, 10);
+
+                const { total_items, total_pages } = await mod.count('id').then(total_items => ({total_items, total_pages: Math.ceil(total_items/page_size_number)}))
+
+                // If the page number is greater than the number of pages, we return an empty array
+                if (page_number > total_pages) {
+                    sendResponse(res, []); // Can I do this directly?
+                    return
+                // If it is the last page, we return only the last elements of the request
+                } else if (page_number === total_pages) {
+                    const left_items = total_items - (page_number-1)*page_size_number;
+
+                    mod.orderBy('id', 'DESC').query(qb => qb.limit(left_items).offset(page_number-1)); // Remove orderBy?
+                // otherwise, return the elements of the page requested
+                } else {
+                    mod.orderBy('id', 'DESC').query(qb => qb.limit(page_size_number).offset(page_number-1)); // Remove orderBy?
+                }
+
+                // Add headers in res with links to previous and next pages
+                // Add also the number of pages and the number of items per page
+                if (!(page_number === 0)) {
+                    res.header('x-prev', `${req.route.path}?page=${page_number - 1}&page_size=${page_size_number}&load=${req.query.load ? req.query.load: ''}`);
```

Comment:
I think the same down here, especially with the long template strings, separate lines would be a lot better.